flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/KeyedStream.java

```diff
@@ -163,7 +166,11 @@ public KeyedStream(DataStream<T> dataStream, KeySelector<T, KEY> keySelector, Ty
 	 * @param size The size of the window.
 	 */
 	public WindowedStream<T, KEY, TimeWindow> timeWindow(AbstractTime size) {
-		return window(TumblingTimeWindows.of(size));
+		if (environment.getStreamTimeCharacteristic() == TimeCharacteristic.PROCESSING_TIME) {
```

Comment:
I'm not sure how the rest of the DataStream API is behaving, but I though that it doesn't matter when the environment settings (for example parallelism) are set.
In this case, users have to set the `TimeCharacteristic` before using operators.